### PR TITLE
OADP-660: Add datamover support for azure blob storage

### PR DESCRIPTION
**Pull request by shubham-pampattiwar**
Related volume-snapshot controller PR: https://github.com/konveyor/volume-snapshot-mover/pull/128

### CODE REVIEW

File: controllers/datamover.go [hunk 1]
```diff
@@ -290,14 +294,82 @@ func (r *DPAReconciler) createResticSecretsPerBSL(dpa *oadpv1alpha1.DataProtecti
...
+			if op == controllerutil.OperationResultCreated || op == controllerutil.OperationResultUpdated {
+				r.EventRecorder.Event(rsecret,
+					corev1.EventTypeNormal,
+					"ResticSecretReconciled",
+					fmt.Sprintf("%s restic secret %s", op, rsecret.Name),
+				)
+			}
+
+		}
 	}
```
Comment (kaovilai) [inline controllers/datamover.go:366]: - Add a case for empty provider
- Add a default case for unsupported providers

**Review by kaovilai — CHANGES_REQUESTED** [review]

File: controllers/datamover.go (hunk 1 above)
Comment (shubham-pampattiwar) [reply controllers/datamover.go:366]: @kaovilai we do not support empty provider or any other provider other than aws/azure/gcp, its already validated in bsl.go https://github.com/openshift/oadp-operator/blob/master/controllers/bsl.go#L61

**Review by savitharaghunathan — COMMENTED** [review]
visual ack
/lgtm

**Review by kaovilai — APPROVED** [review]

**Review by savitharaghunathan — APPROVED** [review]
Tested back/restore workflow and it worked.

**Pull request merged**